plugins/modules/inventory_intent.py

```diff
@@ -2821,6 +2839,42 @@ def get_diff_merged(self, config):
                 self.log(self.msg, "ERROR")
                 return self
 
+        if self.config[0].get('update_interface_details'):
+            device_to_update = self.get_device_ips_from_config_priority()
+            device_exist = self.is_device_exist_for_update(device_to_update)
+
+            if not device_exist:
+                self.msg = """Cannot perform updating the interface details operation as device(s): {0} not present in Cisco
+                            Catalyst Center""".format(str(device_to_update))
+                self.status = "failed"
+                self.result['response'] = self.msg
+                self.log(self.msg, "ERROR")
+                return self
+
+        if self.config[0].get('role'):
+            devices_to_update_role = self.get_device_ips_from_config_priority()
+            device_exist = self.is_device_exist_for_update(devices_to_update_role)
+
+            if not device_exist:
+                self.msg = """Cannot perform Update Device Role operation as device(s): {0} not present in Cisco Catalyst
```

Comment:
Unable to update device role because the device(s) listed: {0} are not present in the Cisco Catalyst Center.

Reply:
added.